app/lib/infrastructure/language_repository.dart

```diff
@@ -2,132 +2,174 @@ part of 'package:word_link/domain/controllers/language_controller.dart';
 
 class _LanguageRepository extends LanguageController {
   @override
-  CollectionObject getMostUsedWords(
-      {required LanguageEnum language, int numberOfWords = 5,}) {
-    final Map<String, String> wordMap = _getWordMapForLanguage(language);
+  Future<CollectionObject> getMostUsedWords({
+    required LanguageEnum language,
+    int numberOfWords = 5,
+  }) async {
+    final List<String> words = _getSourceLanguageWords(language);
 
-    final wordPairs = wordMap.entries
+    final Map<String, String> wordPairs = await _getTranslatedWords(
+        language.translateLang, words.sublist(0, numberOfWords));
+
+    final cards = wordPairs.entries
         .take(numberOfWords)
         .map((entry) => CardObject(name: entry.key, answer: entry.value))
         .toList();
+
     return CollectionObject(
-        name: LanguageEnum.polish.displayName, cardsTemp: wordPairs,);
+      name: LanguageEnum.polish.displayName,
+      cardsTemp: cards,
+    );
   }
 
-  Map<String, String> _getWordMapForLanguage(LanguageEnum language) {
+  List<String> _getSourceLanguageWords(LanguageEnum language) {
     switch (language) {
       case LanguageEnum.polish:
         return polishWords;
     }
   }
+
+  Future<Map<String, String>> _getTranslatedWords(
+      TranslateLanguage language, List<String> words) async {
+    final Map<String, String> wordPairs = {};
+
+    for (final word in words) {
+      final translatedWord =
+          await _translateText(language, TranslateLanguage.english, word);
+      wordPairs[word] = translatedWord;
+    }
+
+    return wordPairs;
+  }
+
+  Future<String> _translateText(TranslateLanguage sourceLanguage,
+      TranslateLanguage targetLanguage, String word) async {
+    final translator = OnDeviceTranslator(
+        sourceLanguage: sourceLanguage, targetLanguage: targetLanguage);
+
+    final modelManager = OnDeviceTranslatorModelManager();
+    final bool isModelDownloaded =
+        await modelManager.isModelDownloaded(targetLanguage.bcpCode);
+
+    if (!isModelDownloaded) {
+      await modelManager.downloadModel(targetLanguage.bcpCode);
+    }
+
+    final String result = await translator.translateText(word);
+
+    translator.close();
+
+    return result;
+  }
 }
 
 enum LanguageEnum {
```

Comment:
This enum is being used directly outside of the repository so it should move to the controller.